### PR TITLE
checks if team exists

### LINKED ISSUE #264: Only Display Team Info if ID Exists
image

- The team only exists if there is a valid ID, otherwise it should ask the user the join a team
- For now, if there is no team, then display nothing

### PR DESCRIPTION
**Pull request by stormyy00**

### CODE REVIEW

File: src/components/User/Dashboard.jsx [hunk 1]
```diff
@@ -101,10 +103,28 @@ const Dashboard = () => {
             setUser={setUser}
             editable={true}
           />
-          <p>TEAM MEMBER 1</p>
+          {/* <p>TEAM MEMBER 1 </p>
```
Comment (shahdivyank) [inline src/components/User/Dashboard.jsx]: remove commented code

File: src/components/User/Dashboard.jsx [hunk 2]
```diff
@@ -101,10 +103,28 @@ const Dashboard = () => {
...
+
           <p>TEAM MEMBER 3</p>
-          <p>TEAM MEMBER 4</p>
+
+          <p>TEAM MEMBER 4</p> */}
+
+          {user.team !== "" ? (
+            <>
+              <p>TEAM MEMBER 1 </p>
+              {teamMembers[0]}
```
Comment (shahdivyank) [inline src/components/User/Dashboard.jsx]: map through the array instead

File: src/components/User/Dashboard.jsx [hunk 3]
```diff
@@ -19,6 +19,8 @@ const Dashboard = () => {
     team: "lololol",
   });
 
+  const teamMembers = ["thing 1", "thing 2", "thing 3", "thing 4"];
```
Comment (shahdivyank) [inline src/components/User/Dashboard.jsx]: use mock data

File: src/components/User/Dashboard.jsx [hunk 4]
```diff
@@ -95,38 +96,47 @@ const Dashboard = () => {
...
-            user={user}
-            setUser={setUser}
-            editable={edit}
-          />
-          <p>TEAM MEMBER 1</p>
-          <p>TEAM MEMBER 2</p>
-          <p>TEAM MEMBER 3</p>
-          <p>TEAM MEMBER 4</p>
+          {user.team ? (
+            <>
```
Comment (shahdivyank) [inline src/components/User/Dashboard.jsx]: extract into component

File: src/components/User/Dashboard.jsx [hunk 5]
```diff
@@ -95,38 +96,47 @@ const Dashboard = () => {
...
+                name="devpost"
+                type="text"
+                title="Devpost Link"
+                value={user.devpost}
+                user={user}
+                setUser={setUser}
+                editable={true}
+              />
+
+              {teamList.slice(0, 4).map((member, index) => (
```
Comment (shahdivyank) [inline src/components/User/Dashboard.jsx]: why is there a slice for size 4?

File: src/components/User/TeamInfo.jsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        value={user.devpost}
+        user={user}
+        editable={true}
+      />
+
+      <div>
+        {team.map((member, memberIndex) => (
+          <div key={memberIndex}>
+            <p>TEAM MEMBER {memberIndex + 1}</p>
+            <p>{member.members[memberIndex]}</p>
```
Comment (shahdivyank) [inline src/components/User/TeamInfo.jsx]: why is this needed? isn't member already an array of members?

Comment (shahdivyank) [reply src/components/User/TeamInfo.jsx]: it should be team.members.map

File: src/components/User/TeamInfo.jsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import Input from "../Input";
+
+const TeamInfo = ({ user, team }) => {
+  return user ? (
```
Comment (shahdivyank) [inline src/components/User/TeamInfo.jsx]: user should always exist?

File: src/components/User/Dashboard.jsx [hunk 8]
```diff
@@ -102,31 +107,14 @@ const Dashboard = () => {
             value={user.team}
             user={user}
             setUser={setUser}
-            editable={edit}
+            editable={true}
```
Comment (shahdivyank) [inline src/components/User/Dashboard.jsx]: this should be based on the editable fields set by the dashboard

File: src/components/User/TeamInfo.jsx (hunk 6 above)
Comment (stormyy00) [reply src/components/User/TeamInfo.jsx]: team.members.map creates an error "Cannot read properties of undefined (reading 'map')"

File: src/components/User/TeamInfo.jsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
+import Input from "../Input";
+
+const TeamInfo = ({ user, team, setUser }) => {
+  return (
+    <div>
```
Comment (shahdivyank) [inline src/components/User/TeamInfo.jsx]: make it a fragment

File: src/components/User/TeamInfo.jsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+      <Input
+        name="devpost"
+        type="text"
+        title="Devpost Link"
+        value={user.devpost}
+        user={user}
+        editable={true}
+      />
+
+      <div>
```
Comment (shahdivyank) [inline src/components/User/TeamInfo.jsx]: is this div needed?

File: src/components/User/UserInfo.jsx [hunk 11]
```diff
@@ -0,0 +1,60 @@
+import { Genders, Grades, Majors } from "@/data/Register";
+import { Schools } from "@/data/Schools";
+import Input from "../Input";
+import Radio from "../Radio";
+import Select from "../Select";
+import Button from "../Forms/Button";
+const UserInfo = ({ handleEdit, handleSave, user, setUser, edit }) => {
```
Comment (shahdivyank) [inline src/components/User/UserInfo.jsx]:
```suggestion

const UserInfo = ({ handleEdit, handleSave, user, setUser, edit }) => {
```

File: src/components/User/TeamInfo.jsx (hunk 10 above)
Comment (stormyy00) [reply src/components/User/TeamInfo.jsx]: no

File: src/components/User/UserInfo.jsx [hunk 12]
```diff
@@ -0,0 +1,60 @@
+import { Genders, Grades, Majors } from "@/data/Register";
+import { Schools } from "@/data/Schools";
+import Input from "../Input";
+import Radio from "../Radio";
+import Select from "../Select";
+import Button from "../Forms/Button";
```
Comment (shahdivyank) [inline src/components/User/UserInfo.jsx]: This button should then be stored in the common folder instead of forms, since its being used here as well

File: src/components/User/UserInfo.jsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+const UserInfo = ({ handleEdit, handleSave, user, setUser, edit }) => {
+  return (
+    <div>
```
Comment (shahdivyank) [inline src/components/User/UserInfo.jsx]: fragment

File: src/components/User/Dashboard.jsx [hunk 14]
```diff
@@ -4,22 +4,13 @@ import ProfileHeader from "./Header";
...
-    phone: "123 123 1234",
-    grade: "undergraduate",
-    school: "samepl scool",
-    major: "nothing im useless",
-    gender: "everythijg",
-    team: "lololol",
-  });
+  const [user, setUser] = useState(mockUser);
   const [edit, setEdit] = useState(false);
   const handleEdit = () => {
```
Comment (shahdivyank) [inline src/components/User/Dashboard.jsx]: if the function only toggles, then we can integrate it directly instead of making a whole function for it

File: src/components/Forms/Mentor.jsx [hunk 15]
```diff
@@ -8,7 +8,7 @@ import Checkbox from "@/components/Checkbox";
 import Input from "@/components/Input";
 import Col from "react-bootstrap/Col";
 import Row from "react-bootstrap/Row";
-import Button from "@/components/Forms/Button";
+import Button from "@/components/Button.jsx";
```
Comment (shahdivyank) [inline src/components/Forms/Mentor.jsx]: remove the JSX

File: src/components/User/Dashboard.jsx [hunk 16]
```diff
@@ -4,50 +4,21 @@ import ProfileHeader from "./Header";
...
-              onClick={handleSave}
-            />
-          ) : (
-            <FaPencil
-              className="hover:cursor-pointer ml-2"
-              onClick={handleEdit}
-            />
-          )}
-        </div>
+        <ProfileHeader email="hello" name="hello" />
```
Comment (shahdivyank) [inline src/components/User/Dashboard.jsx]: should use mock user info

File: src/components/User/TeamInfo.jsx [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+      />
+      <Input
+        name="devpost"
+        type="text"
+        title="Devpost Link"
+        value={user.devpost}
+        user={user}
+        editable={true}
+      />
+
```
Comment (shahdivyank) [inline src/components/User/TeamInfo.jsx]:
```suggestion
```

File: src/data/mock/User.js [hunk 18]
```diff
@@ -0,0 +1,10 @@
+export const mockUser = {
+  phone: "123 123 1234",
+  grade: "undergraduate",
+  school: "samepl scool",
+  major: "nothing im useless",
+  gender: "everythijg",
```
Comment (shahdivyank) [inline src/data/mock/User.js]: do valid values that come from the radio select

File: src/data/mock/Teams.js [hunk 19]
```diff
@@ -0,0 +1,199 @@
+export const teamList = [
```
Comment (shahdivyank) [inline src/data/mock/Teams.js]: pull from dev, these files have been moved to fixtures

Comment (shahdivyank) [reply src/data/mock/Teams.js]: Remove these files

Comment (shahdivyank) [inline src/data/mock/Teams.js]: Remove these mock files

File: src/components/User/TeamInfo.jsx [hunk 20]
```diff
@@ -0,0 +1,41 @@
+import Input from "../Input";
+
+const TeamInfo = ({ user, team, setUser }) => {
```
Comment (shahdivyank) [inline src/components/User/TeamInfo.jsx]: Call it just Team.jsx and User.jsx

File: src/components/Forms/Mentor.jsx (hunk 15 above)
Comment (shahdivyank) [inline src/components/Forms/Mentor.jsx]:
```suggestion
import Button from "@/components/Button";
```

**Review by shahdivyank — APPROVED** [review]

File: src/components/user/Dashboard.jsx [hunk 21]
```diff
@@ -8,46 +8,21 @@ import { GENDERS, GRADES, MAJORS } from "@/data/forms/Information";
...
-              onClick={handleSave}
-            />
-          ) : (
-            <FaPencil
-              className="hover:cursor-pointer ml-2"
-              onClick={handleEdit}
-            />
-          )}
-        </div>
+        <ProfileHeader email="hello" name="hello" />
```
Comment (shahdivyank) [inline src/components/user/Dashboard.jsx]: this should be from mock data

**Review by menthy-wu — APPROVED** [review]
